selendroid-server-common/src/main/java/io/selendroid/server/common/utils/CallLogEntry.java

```diff
@@ -32,6 +48,38 @@ public CallLogEntry(String number, int duration, Date time, int direction) {
         this.direction = direction;
     }
     
+    public static CallLogEntry fromJson(String jsonString) {
+        
+        try {
+            JSONObject json = new JSONObject(jsonString);
+            CallLogEntry entry = new CallLogEntry(json.optString("number"), json.optInt("duration"));
+            entry.direction = json.optInt("direction", INCOMING_TYPE);
+            entry.time = dateFormat.parse(json.optString("time"));
+            return entry;
+        }
+        catch(ParseException e) {
+            throw new IllegalArgumentException("Unable to parse CallLogEntry from string " + jsonString, e);
+            
+        }
+        catch(JSONException e) {
+            throw new IllegalArgumentException("Unable to parse CallLogEntry from string " + jsonString, e);
+        }        
+    }
+    
+    public String toJSON() {
+        try {
+            JSONObject json = new JSONObject();
+            json.put("number", number);
+            json.put("duration", duration);
+            json.put("direction", direction);
+            json.put("time", dateFormat.format(time));
+            return json.toString();
+        }
+        catch(JSONException e) {
```

Comment:
Can you put the catch on the line above? `} catch(...) {` in other places as well.